common/src/main/java/com/wynntils/models/gear/SetModel.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    @Override
+    public void reloadData() {
+        loadSetData();
+    }
+
+    /**
+     * @param setName Name of the set as it appears in game. Eg. "Morph"
+     */
+    public SetInfo getSetInfo(String setName) {
+        return setData.getOrDefault(setName, null);
+    }
+
+    public SetInfo getSetInfoForItem(String itemName) {
+        return getSetInfo(getSetName(itemName));
+    }
+
+    /**
+     * @param itemName The name of the item to check
+     * @return The set name if the item is part of a set, null otherwise
+     */
+    public String getSetName(String itemName) {
+        for (Map.Entry<String, SetInfo> entry : setData.entrySet()) {
+            if (entry.getValue().items().contains(itemName)) {
+                return entry.getKey();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @return A Set of all equipped set names
+     */
+    public Set<String> getUniqueSetNames() {
+        Set<String> returnable = new HashSet<>();
+        for (ItemStack itemStack : Models.PlayerInventory.getEquippedItems()) {
+            Optional<WynnItem> wynnItem = Models.Item.getWynnItem(itemStack);
+            if (wynnItem.isPresent()
+                    && wynnItem.get() instanceof GearItem gear
+                    && gear.getGearTier() == GearTier.SET
+                    && gear.getSetInfo().isPresent()) {
+                returnable.add(gear.getSetInfo().get().name());
+            }
+        }
+        return returnable;
+    }
+
+    public int getTrueCount(String setName) {
+        int trueCount = 0;
+
+        for (ItemStack itemStack : Models.PlayerInventory.getEquippedItems()) {
+            for (StyledText line : LoreUtils.getLore(itemStack)) {
+                Matcher setMatcher = SetModel.SET_PATTERN.matcher(line.getString());
+                if (setMatcher.matches() && setMatcher.group(1).equals(setName)) {
+                    trueCount++;
+                }
+            }
+        }
+
+        return trueCount;
+    }
+
+    public boolean hasSetData() {
+        return !setData.isEmpty();
+    }
+
+    private void loadSetData() {
+        Download dl = Managers.Net.download(UrlId.DATA_STATIC_ITEM_SETS);
+        dl.handleReader(reader -> {
+            TypeToken<Map<String, RawSetInfo>> type = new TypeToken<>() {};
+            Map<String, RawSetInfo> rawSets = Managers.Json.GSON.fromJson(reader, type.getType());
+            rawSets.forEach((setName, rawSetInfo) -> {
+                List<Map<StatType, Integer>> bonuses = rawSetInfo.bonuses.stream()
+                        .map(bonusPair -> {
+                            Map<StatType, Integer> bonusMap = new HashMap<>();
+                            bonusPair.forEach((statName, statValue) -> {
+                                StatType statType = Models.Stat.fromApiName(statName);
+                                if (statType == null) {
+                                    WynntilsMod.warn("Unknown stat type: " + statName);
```

Comment:
Shouldn't you put a continue here? I don't think null should make it into the bonus map.